lib/modules/manager/gomod/extract.ts

```diff
@@ -70,9 +70,13 @@ export function extractPackageFile(content: string): PackageFile | null {
         deps.push(dep);
       }
       const requireMatch = regEx(/^require\s+([^\s]+)\s+([^\s]+)/).exec(line);
-      if (requireMatch && !line.endsWith('// indirect')) {
-        logger.trace({ lineNumber }, `require line: "${line}"`);
-        const dep = getDep(lineNumber, requireMatch, 'require');
+      if (requireMatch) {
+        const requireType = line.endsWith('// indirect')
+          ? 'indirect'
+          : 'require';
+
+        logger.trace({ lineNumber }, `${requireType} line: "${line}"`);
+        const dep = getDep(lineNumber, requireMatch, requireType);
```

Comment:
set `enabled=false` flag, so we can later use package rules to override by user

Reply:
Or are you thinking to add that in as a new field?

Reply:
Maybe something like?

```typescript
          dep.managerData!.enabled = false

```

Reply:
no, it should be added to `Package<T>` interface

Reply:
Sure! Happy for it to be in this PR or shall I do a separate one?

Reply:
this one